crates/sui-graphql-rpc/src/server/sui_sdk_data_provider.rs

```diff
@@ -572,44 +575,6 @@ pub(crate) fn convert_to_validators(
     Ok(result)
 }
 
-async fn convert_to_transaction_block(
-    cl: &SuiClient,
-    tx: sui_json_rpc_types::SuiTransactionBlockResponse,
-) -> Result<TransactionBlock> {
-    let tx_data = tx.transaction.as_ref().unwrap();
-    let tx_effects = tx.effects.as_ref().unwrap();
-    let sender = *tx_data.data.sender();
-    let gas_effects = GasEffects::new(tx_effects.gas_cost_summary(), tx_effects.gas_object());
-    let gcs = tx_effects.gas_cost_summary();
-    let system_state = cl.governance_api().get_latest_sui_system_state().await?;
-    let protocol_configs = cl.fetch_protocol_config(None).await?;
-    let epoch = convert_to_epoch(gcs, &system_state, &protocol_configs)?;
-    let txd = TransactionDigest::from_array(tx.digest.into_inner());
-    let expiration = epoch.clone();
-    Ok(TransactionBlock {
-        digest: txd,
-        effects: Some(TransactionBlockEffects {
-            digest: txd,
```

Comment:
@wlmyng your question about redundancies in the schema got me curious, so I had a look here.

This digest is not redundant, it's just filled in incorrectly here.  This should be the **effects digest**, not the transaction digest as it is now (i.e. this is a bug).

We shouldn't change the schema here -- in the protocol, the transaction and its effects are distinct structures, with distinct IDs (digests).

Reply:
I see, that makes a lot of sense 😅